src/addressResolvers/farcaster.ts

```diff
@@ -0,0 +1,107 @@
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import { graphQlCall, Address, Handle, FetchError, isSilencedError, isEvmAddress } from './utils';
+
+export const NAME = 'Farcaster';
+const FNAMES_API_URL = 'https://fnames.farcaster.xyz/transfers?name=';
+const NEYNAR_API_URL = 'https://api.neynar.com/v2/farcaster/user/';
+const API_KEY = process.env.NEYNAR_API_KEY ?? '';
+
+
+interface UserDetails {
+  username: string;
+  verified_addresses: {
+    eth_addresses: string[];
+    sol_addresses: string[];
+  };
+  pfp_url: string;
+}
+
+interface ApiResponse {
+  [address: string]: UserDetails[];
+}
+
+interface UserResult {
+  username?: string;
+  eth_addresses?: string[];
+  sol_addresses?: string[];
+  pfp_url?: string;
+}
+
+
+async function fetchData<T>(url: string, method: string = 'GET'): Promise<T> {
+  const headers = {
+    Accept: 'application/json',
+    api_key: API_KEY
+  };
+  const response = await fetch(url, { method, headers });
+  if (!response.ok) {
+    throw new Error(`Failed to fetch data from the API. Status: ${response.status}`);
+  }
+  return response.json() as Promise<T>;
+}
+
+export async function lookupAddresses(
```

Comment:
Function signature is 

```
export async function lookupAddresses(addresses: Address[]): Promise<Record<Address, Handle>> {
```

Reply:
Also, can you move this after `fetchUserDetailsByUsername` function, so all exported functions are together ?